Have not added the contract itself

First, the recommended `npm run build-contracts` command has failed:

```
$ npm run build-contracts

> ERC777iHomeCredits@1.0.0 build-contracts /home/olegabr/work/ERC777iHomeCredits
> sol-merger "./contracts/*.sol" ./build

,
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ERC777iHomeCredits@1.0.0 build-contracts: `sol-merger "./contracts/*.sol" ./build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ERC777iHomeCredits@1.0.0 build-contracts script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/olegabr/.npm/_logs/2019-10-15T10_27_23_541Z-debug.log
```

I've worked it around with:

```
./node_modules/sol-merger/dist/bin/sol-merger.js "./contracts/*.sol" ./build
```

and it produces file in the build directory, and this file included all but the `ERC777iHomeCredits` contract definition. I've added it manually and verification succeeded.

--- a/bin/sol-merger.ts
+++ b/bin/sol-merger.ts
@@ -78,6 +78,6 @@ async function execute(err: Error, files: string[]) {
   });
 
   Promise.all(promises)
-    .then(done)
+    .then(() => done())
     .catch(done);
 }

--- a/lib/fileAnalyzer.ts
+++ b/lib/fileAnalyzer.ts
@@ -129,11 +129,16 @@ export class FileAnalyzer {
    *
    */
   analyzeExports(contents: string): FileAnalyzerExportsResult[] {
-    const exportRegex = /(contract|library|interface)\s+([a-zA-Z_$][a-zA-Z_$0-9]*)\s*([^"]*?)\{/g;
+    const exportRegex = /(contract|library|interface)\s+([a-zA-Z_$][a-zA-Z_$0-9]*)\s*([\s\S]*?)\{/g;
+    const isRegex = /^is\s*[a-zA-Z_$][a-zA-Z_$0-9]*(.[a-zA-Z_$][a-zA-Z_$0-9]*)?(\([\s\S]*?\))?(,\s*?[a-zA-Z_$][a-zA-Z_$0-9]*(.[a-zA-Z_$][a-zA-Z_$0-9]*)?(\([\s\S]*?\))?)?\s*$/;
     const results = [];
     let group: RegExpExecArray;
     while ((group = exportRegex.exec(contents))) {
       const [, type, name, is] = group;
+      // Checking that `is` clause is correct
+      if (is.trim() && !isRegex.test(is.trim())) {
+        continue;
+      }
       const body = this.findBodyEnd(
         contents,
         group.index + group[0].length - 1,

--- a/package.json
+++ b/package.json
@@ -56,6 +56,7 @@
     "@types/strip-json-comments": "0.0.30",
     "chai": "^4.2.0",
     "mocha": "^6.1.4",
+    "openzeppelin-solidity": "^2.3.0",
     "ts-node": "^8.3.0",
     "tslib": "^1.10.0",
     "tslint": "^5.17.0",

--- a/test/compiled/DefaultParamentersInheritance.sol
+++ b/test/compiled/DefaultParamentersInheritance.sol
@@ -1,0 +1,562 @@
+pragma solidity ^0.5.1;
+
+
+interface IERC777 {
+    
+    function name() external view returns (string memory);
+
+    
+    function symbol() external view returns (string memory);
+
+    
+    function granularity() external view returns (uint256);
+
+    
+    function totalSupply() external view returns (uint256);
+
+    
+    function balanceOf(address owner) external view returns (uint256);
+
+    
+    function send(address recipient, uint256 amount, bytes calldata data) external;
+
+    
+    function burn(uint256 amount, bytes calldata data) external;
+
+    
+    function isOperatorFor(address operator, address tokenHolder) external view returns (bool);
+
+    
+    function authorizeOperator(address operator) external;
+
+    
+    function revokeOperator(address operator) external;
+
+    
+    function defaultOperators() external view returns (address[] memory);
+
+    
+    function operatorSend(
+        address sender,
+        address recipient,
+        uint256 amount,
+        bytes calldata data,
+        bytes calldata operatorData
+    ) external;
+
+    
+    function operatorBurn(
+        address account,
+        uint256 amount,
+        bytes calldata data,
+        bytes calldata operatorData
+    ) external;
+
+    event Sent(
+        address indexed operator,
+        address indexed from,
+        address indexed to,
+        uint256 amount,
+        bytes data,
+        bytes operatorData
+    );
+
+    event Minted(address indexed operator, address indexed to, uint256 amount, bytes data, bytes operatorData);
+
+    event Burned(address indexed operator, address indexed from, uint256 amount, bytes data, bytes operatorData);
+
+    event AuthorizedOperator(address indexed operator, address indexed tokenHolder);
+
+    event RevokedOperator(address indexed operator, address indexed tokenHolder);
+}
+
+interface IERC777Recipient {
+    
+    function tokensReceived(
+        address operator,
+        address from,
+        address to,
+        uint amount,
+        bytes calldata userData,
+        bytes calldata operatorData
+    ) external;
+}
+
+interface IERC777Sender {
+    
+    function tokensToSend(
+        address operator,
+        address from,
+        address to,
+        uint amount,
+        bytes calldata userData,
+        bytes calldata operatorData
+    ) external;
+}
+
+interface IERC20 {
+    
+    function totalSupply() external view returns (uint256);
+
+    
+    function balanceOf(address account) external view returns (uint256);
+
+    
+    function transfer(address recipient, uint256 amount) external returns (bool);
+
+    
+    function allowance(address owner, address spender) external view returns (uint256);
+
+    
+    function approve(address spender, uint256 amount) external returns (bool);
+
+    
+    function transferFrom(address sender, address recipient, uint256 amount) external returns (bool);
+
+    
+    event Transfer(address indexed from, address indexed to, uint256 value);
+
+    
+    event Approval(address indexed owner, address indexed spender, uint256 value);
+}
+
+library SafeMath {
+    
+    function add(uint256 a, uint256 b) internal pure returns (uint256) {
+        uint256 c = a + b;
+        require(c >= a, "SafeMath: addition overflow");
+
+        return c;
+    }
+
+    
+    function sub(uint256 a, uint256 b) internal pure returns (uint256) {
+        require(b <= a, "SafeMath: subtraction overflow");
+        uint256 c = a - b;
+
+        return c;
+    }
+
+    
+    function mul(uint256 a, uint256 b) internal pure returns (uint256) {
+        
+        
+        
+        if (a == 0) {
+            return 0;
+        }
+
+        uint256 c = a * b;
+        require(c / a == b, "SafeMath: multiplication overflow");
+
+        return c;
+    }
+
+    
+    function div(uint256 a, uint256 b) internal pure returns (uint256) {
+        
+        require(b > 0, "SafeMath: division by zero");
+        uint256 c = a / b;
+        
+
+        return c;
+    }
+
+    
+    function mod(uint256 a, uint256 b) internal pure returns (uint256) {
+        require(b != 0, "SafeMath: modulo by zero");
+        return a % b;
+    }
+}
+
+library Address {
+    
+    function isContract(address account) internal view returns (bool) {
+        
+        
+        
+
+        uint256 size;
+        
+        assembly { size := extcodesize(account) }
+        return size > 0;
+    }
+}
+
+interface IERC1820Registry {
+    
+    function setManager(address account, address newManager) external;
+
+    
+    function getManager(address account) external view returns (address);
+
+    
+    function setInterfaceImplementer(address account, bytes32 interfaceHash, address implementer) external;
+
+    
+    function getInterfaceImplementer(address account, bytes32 interfaceHash) external view returns (address);
+
+    
+    function interfaceHash(string calldata interfaceName) external pure returns (bytes32);
+
+    
+    function updateERC165Cache(address account, bytes4 interfaceId) external;
+
+    
+    function implementsERC165Interface(address account, bytes4 interfaceId) external view returns (bool);
+
+    
+    function implementsERC165InterfaceNoCache(address account, bytes4 interfaceId) external view returns (bool);
+
+    event InterfaceImplementerSet(address indexed account, bytes32 indexed interfaceHash, address indexed implementer);
+
+    event ManagerChanged(address indexed account, address indexed newManager);
+}
+
+contract ERC777 is IERC777, IERC20 {
+    using SafeMath for uint256;
+    using Address for address;
+
+    IERC1820Registry private _erc1820 = IERC1820Registry(0x1820a4B7618BdE71Dce8cdc73aAB6C95905faD24);
+
+    mapping(address => uint256) private _balances;
+
+    uint256 private _totalSupply;
+
+    string private _name;
+    string private _symbol;
+
+    
+    
+
+    
+    bytes32 constant private TOKENS_SENDER_INTERFACE_HASH =
+        0x29ddb589b1fb5fc7cf394961c1adf5f8c6454761adf795e67fe149f658abe895;
+
+    
+    bytes32 constant private TOKENS_RECIPIENT_INTERFACE_HASH =
+        0xb281fc8c12954d22544db45de3159a39272895b169a852b314f9cc762e44c53b;
+
+    
+    address[] private _defaultOperatorsArray;
+
+    
+    mapping(address => bool) private _defaultOperators;
+
+    
+    mapping(address => mapping(address => bool)) private _operators;
+    mapping(address => mapping(address => bool)) private _revokedDefaultOperators;
+
+    
+    mapping (address => mapping (address => uint256)) private _allowances;
+
+    
+    constructor(
+        string memory name,
+        string memory symbol,
+        address[] memory defaultOperators
+    ) public {
+        _name = name;
+        _symbol = symbol;
+
+        _defaultOperatorsArray = defaultOperators;
+        for (uint256 i = 0; i < _defaultOperatorsArray.length; i++) {
+            _defaultOperators[_defaultOperatorsArray[i]] = true;
+        }
+
+        
+        _erc1820.setInterfaceImplementer(address(this), keccak256("ERC777Token"), address(this));
+        _erc1820.setInterfaceImplementer(address(this), keccak256("ERC20Token"), address(this));
+    }
+
+    
+    function name() public view returns (string memory) {
+        return _name;
+    }
+
+    
+    function symbol() public view returns (string memory) {
+        return _symbol;
+    }
+
+    
+    function decimals() public pure returns (uint8) {
+        return 18;
+    }
+
+    
+    function granularity() public view returns (uint256) {
+        return 1;
+    }
+
+    
+    function totalSupply() public view returns (uint256) {
+        return _totalSupply;
+    }
+
+    
+    function balanceOf(address tokenHolder) public view returns (uint256) {
+        return _balances[tokenHolder];
+    }
+
+    
+    function send(address recipient, uint256 amount, bytes calldata data) external {
+        _send(msg.sender, msg.sender, recipient, amount, data, "", true);
+    }
+
+    
+    function transfer(address recipient, uint256 amount) external returns (bool) {
+        require(recipient != address(0), "ERC777: transfer to the zero address");
+
+        address from = msg.sender;
+
+        _callTokensToSend(from, from, recipient, amount, "", "");
+
+        _move(from, from, recipient, amount, "", "");
+
+        _callTokensReceived(from, from, recipient, amount, "", "", false);
+
+        return true;
+    }
+
+    
+    function burn(uint256 amount, bytes calldata data) external {
+        _burn(msg.sender, msg.sender, amount, data, "");
+    }
+
+    
+    function isOperatorFor(
+        address operator,
+        address tokenHolder
+    ) public view returns (bool) {
+        return operator == tokenHolder ||
+            (_defaultOperators[operator] && !_revokedDefaultOperators[tokenHolder][operator]) ||
+            _operators[tokenHolder][operator];
+    }
+
+    
+    function authorizeOperator(address operator) external {
+        require(msg.sender != operator, "ERC777: authorizing self as operator");
+
+        if (_defaultOperators[operator]) {
+            delete _revokedDefaultOperators[msg.sender][operator];
+        } else {
+            _operators[msg.sender][operator] = true;
+        }
+
+        emit AuthorizedOperator(operator, msg.sender);
+    }
+
+    
+    function revokeOperator(address operator) external {
+        require(operator != msg.sender, "ERC777: revoking self as operator");
+
+        if (_defaultOperators[operator]) {
+            _revokedDefaultOperators[msg.sender][operator] = true;
+        } else {
+            delete _operators[msg.sender][operator];
+        }
+
+        emit RevokedOperator(operator, msg.sender);
+    }
+
+    
+    function defaultOperators() public view returns (address[] memory) {
+        return _defaultOperatorsArray;
+    }
+
+    
+    function operatorSend(
+        address sender,
+        address recipient,
+        uint256 amount,
+        bytes calldata data,
+        bytes calldata operatorData
+    )
+    external
+    {
+        require(isOperatorFor(msg.sender, sender), "ERC777: caller is not an operator for holder");
+        _send(msg.sender, sender, recipient, amount, data, operatorData, true);
+    }
+
+    
+    function operatorBurn(address account, uint256 amount, bytes calldata data, bytes calldata operatorData) external {
+        require(isOperatorFor(msg.sender, account), "ERC777: caller is not an operator for holder");
+        _burn(msg.sender, account, amount, data, operatorData);
+    }
+
+    
+    function allowance(address holder, address spender) public view returns (uint256) {
+        return _allowances[holder][spender];
+    }
+
+    
+    function approve(address spender, uint256 value) external returns (bool) {
+        address holder = msg.sender;
+        _approve(holder, spender, value);
+        return true;
+    }
+
+   
+    function transferFrom(address holder, address recipient, uint256 amount) external returns (bool) {
+        require(recipient != address(0), "ERC777: transfer to the zero address");
+        require(holder != address(0), "ERC777: transfer from the zero address");
+
+        address spender = msg.sender;
+
+        _callTokensToSend(spender, holder, recipient, amount, "", "");
+
+        _move(spender, holder, recipient, amount, "", "");
+        _approve(holder, spender, _allowances[holder][spender].sub(amount));
+
+        _callTokensReceived(spender, holder, recipient, amount, "", "", false);
+
+        return true;
+    }
+
+    
+    function _mint(
+        address operator,
+        address account,
+        uint256 amount,
+        bytes memory userData,
+        bytes memory operatorData
+    )
+    internal
+    {
+        require(account != address(0), "ERC777: mint to the zero address");
+
+        
+        _totalSupply = _totalSupply.add(amount);
+        _balances[account] = _balances[account].add(amount);
+
+        _callTokensReceived(operator, address(0), account, amount, userData, operatorData, true);
+
+        emit Minted(operator, account, amount, userData, operatorData);
+        emit Transfer(address(0), account, amount);
+    }
+
+    
+    function _send(
+        address operator,
+        address from,
+        address to,
+        uint256 amount,
+        bytes memory userData,
+        bytes memory operatorData,
+        bool requireReceptionAck
+    )
+        private
+    {
+        require(from != address(0), "ERC777: send from the zero address");
+        require(to != address(0), "ERC777: send to the zero address");
+
+        _callTokensToSend(operator, from, to, amount, userData, operatorData);
+
+        _move(operator, from, to, amount, userData, operatorData);
+
+        _callTokensReceived(operator, from, to, amount, userData, operatorData, requireReceptionAck);
+    }
+
+    
+    function _burn(
+        address operator,
+        address from,
+        uint256 amount,
+        bytes memory data,
+        bytes memory operatorData
+    )
+        private
+    {
+        require(from != address(0), "ERC777: burn from the zero address");
+
+        _callTokensToSend(operator, from, address(0), amount, data, operatorData);
+
+        
+        _totalSupply = _totalSupply.sub(amount);
+        _balances[from] = _balances[from].sub(amount);
+
+        emit Burned(operator, from, amount, data, operatorData);
+        emit Transfer(from, address(0), amount);
+    }
+
+    function _move(
+        address operator,
+        address from,
+        address to,
+        uint256 amount,
+        bytes memory userData,
+        bytes memory operatorData
+    )
+        private
+    {
+        _balances[from] = _balances[from].sub(amount);
+        _balances[to] = _balances[to].add(amount);
+
+        emit Sent(operator, from, to, amount, userData, operatorData);
+        emit Transfer(from, to, amount);
+    }
+
+    function _approve(address holder, address spender, uint256 value) private {
+        
+        
+        
+        require(spender != address(0), "ERC777: approve to the zero address");
+
+        _allowances[holder][spender] = value;
+        emit Approval(holder, spender, value);
+    }
+
+    
+    function _callTokensToSend(
+        address operator,
+        address from,
+        address to,
+        uint256 amount,
+        bytes memory userData,
+        bytes memory operatorData
+    )
+        private
+    {
+        address implementer = _erc1820.getInterfaceImplementer(from, TOKENS_SENDER_INTERFACE_HASH);
+        if (implementer != address(0)) {
+            IERC777Sender(implementer).tokensToSend(operator, from, to, amount, userData, operatorData);
+        }
+    }
+
+    
+    function _callTokensReceived(
+        address operator,
+        address from,
+        address to,
+        uint256 amount,
+        bytes memory userData,
+        bytes memory operatorData,
+        bool requireReceptionAck
+    )
+        private
+    {
+        address implementer = _erc1820.getInterfaceImplementer(to, TOKENS_RECIPIENT_INTERFACE_HASH);
+        if (implementer != address(0)) {
+            IERC777Recipient(implementer).tokensReceived(operator, from, to, amount, userData, operatorData);
+        } else if (requireReceptionAck) {
+            require(!to.isContract(), "ERC777: token recipient contract has no implementer for ERC777TokensRecipient");
+        }
+    }
+}
+
+contract ERC777Test is ERC777(
+  "Test",
+  "TSX",
+  new address[](0)
+) {
+  constructor() public {
+    _mint(
+      msg.sender,
+      0x476Bb28Bc6D0e9De04dB5E19912C392F9a76535d,
+      100000000000 * 10 ** uint256(decimals()),
+      "",
+      ""
+    );
+  }
+}

--- a/test/contracts/DefaultParamentersInheritance.sol
+++ b/test/contracts/DefaultParamentersInheritance.sol
@@ -1,0 +1,19 @@
+pragma solidity ^0.5.1;
+
+import "openzeppelin-solidity/contracts/token/ERC777/ERC777.sol";
+
+contract ERC777Test is ERC777(
+  "Test",
+  "TSX",
+  new address[](0)
+) {
+  constructor() public {
+    _mint(
+      msg.sender,
+      0x476Bb28Bc6D0e9De04dB5E19912C392F9a76535d,
+      100000000000 * 10 ** uint256(decimals()),
+      "",
+      ""
+    );
+  }
+}

--- a/test/index.spec.ts
+++ b/test/index.spec.ts
@@ -51,4 +51,8 @@ describe('Solidity Merger', () => {
   it('should not find keywords in strings', async () => {
     await testFile('ReservedWordsInString');
   });
+
+  it('should compile extended class with default parameters', async () => {
+    await testFile('DefaultParamentersInheritance')
+  })
 });

